src/linux_udev.rs

```diff
@@ -0,0 +1,762 @@
+//! Functions for talking to udev directly instead of going through hidapi
```

Comment:
Probably not a good name for the backend.

I would just call it linux.rs, or maybe linux_native.rs?